runners/core-java/build.gradle

```diff
@@ -45,6 +45,6 @@ dependencies {
   shadowTest library.java.mockito_core
   shadowTest library.java.junit
   shadowTest library.java.slf4j_api
-  shadowTest library.java.slf4j_simple
   shadowTest library.java.jackson_dataformat_yaml
+  testRuntimeOnly library.java.slf4j_simple
```

Comment:
I wonder, is `testRuntimeOnly` safe because it circumvents the shading? I suppose we do not shade `sl4j_simple`.

Reply:
Seems so. In any case is as you said, we should not shade it just guarantee is available at runtime.

Reply:
Right now most modules don't set [testShadowJar](https://github.com/apache/beam/blob/a6a6b8efcbd000299e1ffb808eae131156198455/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L96) to be true. There was some work to make a few modules test that way but it hasn't been done for the rest. I think people assumed that vendoring would replace shading everywhere (but vendoring work has stalled) so testing with the shadow jar stalled.

You can use the [shadowTestRuntimeClasspath](https://github.com/apache/beam/blob/a6a6b8efcbd000299e1ffb808eae131156198455/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L917) for projects which run tests using the shadowTest jar. Otherwise the testRuntimeOnly is the right place to do this. (Should we rename shadowTestRuntimeClasspath to be shadowTestRuntimeOnly to match the naming convention within Gradle already?)

Reply:
Thanks for all this information. i was not at all aware of such functionalities. I agree the renaming makes sense but let's do that in a different PR.
Will modules using testShadowJar be tested using the 'shaded' jar? If yes this is great!

Reply:
Yes, modules that set testShadowJar will run their unit tests using the shaded jars

Reply:
This is great improvement, I will try to check in which modules it is worth to do the switch (in particular the ones who do shade stuff to enable it.